internal/pkg/agent/application/pipeline/actions/handlers/handler_action_policy_change.go

```diff
@@ -142,30 +143,50 @@ func (h *PolicyChange) handleFleetServerHosts(ctx context.Context, c *config.Con
 			err, "fail to create API client with updated hosts",
 			errors.TypeNetwork, errors.M("hosts", h.config.Fleet.Client.Hosts))
 	}
+
 	ctx, cancel := context.WithTimeout(ctx, apiStatusTimeout)
 	defer cancel()
-	resp, err := client.Send(ctx, "GET", "/api/status", nil, nil, nil)
+
+	// The new client might have several fleet hosts, we need to try all of them before failing.
+	// As it's a new client, just trying
+	hostsCount := len(h.config.Fleet.Client.Hosts)
+	var resp *http.Response
+	for i := 0; i < hostsCount; i++ {
```

Comment:
bringing slack discussion here:
what would you say about preferring last working fleet server first to save roundtrips.

there's also concern about making last working a bottleneck,
i think current approach makes first one bottleneck as all instances are trying to check it first. 
maybe we could just cache hosts and reshuffle on failed. e.g
if you have hosts 1,2,3,4 and 1st and 2nd fails then you create a new cached hosts array with 3,4,1,2
this would be less a bottleneck as last working may be different for every instance.

Reply:
I ended up sorting the list instead of making it into a queue. A Queue is a good idea though. However as it's local to each agent, and a rather small list I don't think the refactor is worth it now